### PR TITLE
Added reporting to Testlink XML

### PR DESCRIPTION
**Pull request by ConorPKeegan**
# Changes
* Added exporter to produce report for Testlink in XML format
* Added command line switch `--report-testlink-xml` to output the Testlink XML format to a file

### CODE REVIEW

**Comment by ConorPKeegan** [comment]
@mazimkhan @PrzemekWirkus FYI

File: mbed_greentea/mbed_report_api.py [hunk 1]
```diff
@@ -217,6 +217,48 @@ def exporter_testcase_junit(test_result_ext, test_suite_properties=None):
...
+    time = 0.0
+    for target_toolcahin, testsuites in test_result_ext.iteritems():
+        for testsuite_name, testsuite in testsuites.iteritems():
+            time += testsuite['elapsed_time']
+            tests += len(testsuite['testcase_result'])
+            for testcase_name, testcase in testsuite['testcase_result'].iteritems():
+                if testcase['result'] < 0:
+                    errors += 1
+                failures += testcase['failed']
+    generator.startElement('testsuite', AttributesImpl({'errors': str(errors), 'failures': str(failures), 'tests': str(tests), 'time': str(time)}))
```
Comment (PrzemekWirkus) [inline mbed_greentea/mbed_report_api.py]: Please keep lines shorter. You can wrap lines like this:
```python
generator.startElement('testsuite', AttributesImpl({
    'errors': str(errors), 
    'failures': str(failures), 
    'tests': str(tests), 
    'time': str(time)}))
```

File: mbed_greentea/mbed_report_api.py [hunk 2]
```diff
@@ -217,6 +217,48 @@ def exporter_testcase_junit(test_result_ext, test_suite_properties=None):
...
+
+    for target_toolcahin, testsuites in test_result_ext.iteritems():
+        for testsuite_name, testsuite in testsuites.iteritems():
+            for testcase_name, testcase in testsuite['testcase_result'].iteritems():
+                output.write("\n\t".expandtabs(4))
+                generator.startElement('testcase', AttributesImpl({'classname': testcase_name, 'host_os': host_os, 'name': testcase_name, 'time': str(testcase['duration'])}))
```
Comment (PrzemekWirkus) [inline mbed_greentea/mbed_report_api.py]: Please keep this line shorter too :)

**Comment by ConorPKeegan** [comment]
See #175 for replacement PR

**Pull request closed without merging**